### PR TITLE
Fix codename detection

### PR DESCRIPTION
**Pull request by pbruski**

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/Docker.kt [hunk 1]
```diff
@@ -19,14 +19,13 @@ internal class Docker {
             packages = listOf(
                 "apt-transport-https",
                 "ca-certificates",
-                "curl",
-                "software-properties-common"
```
Comment (pbruski) [inline src/main/kotlin/com/atlassian/performance/tools/infrastructure/Docker.kt:23]: this is 150MBs to get one line for a text file...

File: src/test/kotlin/com/atlassian/performance/tools/infrastructure/DockerIT.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.junit.Test
+import java.time.Duration
+
+
+class DockerIT {
+    @Test
+    fun installWorks() {
+        SshUbuntuContainer().start().use { ssh ->
+            ssh.toSsh().newConnection().use { connection ->
+                //workaround for a bug in Docker download site for bionic
```
Comment (pbruski) [inline src/test/kotlin/com/atlassian/performance/tools/infrastructure/DockerIT.kt:13]: This sucks. I think we should have a helper class that sets up this container from a pristine ubuntu container. I'll spike it.

**Comment by mzyromski-atlassian** [comment]
@pbruski, could you please add a CHANGELOG entry?

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ Dropping a requirement of a major version of a dependency is a new contract.
 
 ### Fixed
 - Eliminate spurious apt warnings. Fix [JPERF-642](https://ecosystem.atlassian.net/browse/JPERF-642).
+- Corrected Ubuntu codename detection
```
Comment (wyrzyk) [inline CHANGELOG.md:35]: Can you also take care of the commit message?

Comment (mzyromski-atlassian) [reply CHANGELOG.md:35]: Also, sorry for that, but in the meantime #70 was merged and this entry should go into Unreleased section. Could you please rebase?

**Comment by mzyromski-atlassian** [comment]
Released in `4.16.1`.

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/Docker.kt (hunk 1 above)
Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/infrastructure/Docker.kt:23]: Wow! Great catch

File: src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/os/Ubuntu.kt [hunk 4]
```diff
@@ -100,4 +100,8 @@ class Ubuntu {
         ssh.execute("echo '$repository' | sudo tee /etc/apt/sources.list.d/${sourceFileName}.list")
         updatePackageIndex(ssh)
     }
+
+    fun getDistributionCodename(ssh: SshConnection): String {
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/os/Ubuntu.kt:104]: New API FYI @mzyromski-atlassian

**Comment by dagguh** [comment]
> Released in `4.16.1`.

Should be `4.17.0`, new API was added.

**Comment by mzyromski-atlassian** [comment]
> Should be `4.17.0`, new API was added.

True, missed that. :( Would you advise releasing `4.17.0` now?

**Comment by dagguh** [comment]
Yes, it will allow the new API consumers to depend on 4.17.0+ and still being able to downgrade patch versions in peace.

**Pull request merged**